### PR TITLE
Add missing logs and argument validations.

### PR DESCRIPTION
**Pull request by leegeth**
*Description of changes:*
Add missing logs and argument validations.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by leegeth** [comment]
> The complexity of `MQTT_SerializeConnect` increased from 8 to 17. Please fix

Fixed.

File: libraries/standard/mqtt/src/mqtt.c [hunk 1]
```diff
@@ -291,15 +355,24 @@ MQTTStatus_t MQTT_Disconnect( MQTTContext_t * const pContext )
         bytesSent = sendPacket( pContext,
                                 pContext->networkBuffer.pBuffer,
                                 packetSize );
+        IotLogDebugWithArgs( "Sent %d bytes of disconnect packet. ",
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt.c:358]: This log seems duplicated in the else part.

File: libraries/standard/mqtt/src/mqtt.c [hunk 2]
```diff
@@ -120,34 +153,50 @@ MQTTStatus_t MQTT_Connect( MQTTContext_t * const pContext,
         bytesSent = sendPacket( pContext,
                                 pContext->networkBuffer.pBuffer,
                                 packetSize );
+        IotLogDebugWithArgs( "Sent %d bytes of CONNECT packet. ",
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt.c:156]: This log seems duplicated in the else below.

File: libraries/standard/mqtt/src/mqtt.c (hunk 1 above)
Comment (leegeth) [reply libraries/standard/mqtt/src/mqtt.c]: Thanks. Fixed it

File: libraries/standard/mqtt/src/mqtt.c (hunk 2 above)
Comment (leegeth) [reply libraries/standard/mqtt/src/mqtt.c]: Thanks. Fixed it

**Pull request merged**